cmd/juju/action/run_test.go

```diff
@@ -108,6 +108,11 @@ func (s *RunSuite) TestInit(c *gc.C) {
 		should:      "fail with wrong formatting of k-v args",
 		args:        []string{validUnitId, "valid-action-name", "no-go?od=3"},
 		expectError: "key \"no-go\\?od\" must start and end with lowercase alphanumeric, and contain only lowercase alphanumeric and hyphens",
+	}, {
+		should:       "work with numeric values",
+		args:         []string{validUnitId, "action-01"},
```

Comment:
Maybe a few more test cases to cover things like "-not-valid", "hyphen-at-end-", "9-validrule", others? - based on expected Error "must start and end with lowercase alphanumeric, and contain only lowercase alphanumeric and hyphens"